goss-testing/tests/ncn/goss-iuf-activity-abort.yaml

```diff
@@ -0,0 +1,62 @@
+#
+# MIT License
+#
+# (C) Copyright 2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+{{ $pre_checks := .Vars.iuf_pre_checks }}
+{{ $scripts := .Env.GOSS_BASE | printf "%s/scripts" }}
+{{ $media_dir := .Env.GOSS_BASE | printf "%s/scripts/iuf_run_setup" }}
+{{ $activity := .Vars.iuf_activity }}
+{{ $logrun := .Env.GOSS_BASE | printf "%s/scripts/log_run.sh" }}
+{{ $iuf_pre_checks := $scripts | printf "%s/python/iuf_pre_checks" }}
+{{ $iuf_run := $scripts | printf "%s/python/iuf_run" }}
+{{ $iuf_abort := $scripts | printf "%s/check_iuf_abort.sh" }}
+{{ $iuf_cleanup := $scripts | printf "%s/python/iuf_cleanup" }}
+command:
+  {{ $testlabel := "iuf_activity_abort" }}
+  {{$testlabel}}:
+    title: iuf activity abort
+    meta:
+      desc: Executes Abort for a particular IUF activity.
+      sev: 0
+    exec: |-
+      "{{$logrun}}" -l "{{$testlabel}}" bash -c '
+        # Run pre-checks for IUF
+        "{{$iuf_pre_checks}}" "{{ index $pre_checks 0 }}" "{{ index $pre_checks 1 }}" "{{ index $pre_checks 2 }}" "{{ index $pre_checks 3 }}" "{{ index $pre_checks 4 }}" "{{ index $pre_checks 5 }}" && \
+
+        # Run IUF abort
+        "{{$iuf_abort}}"
+
+        # Capture the exit code of the previous step
+        run_exit_code=$?
+
+        # Run cleanup step independently
+        "{{$iuf_cleanup}}" "{{ index $activity 0 }}"
+
+        # Exit with code 1 if the iuf abort command failed with exit code 1
+        if [ $run_exit_code -eq 1 ]; then

```

Comment:
Shouldn't the test fail if iuf_abort returned any non-0 exit code? Or are some non-0 returns expected and not a problem? Same comment for the other tests.